Jenkinsfile

```diff
@@ -1,23 +1,51 @@
-node {
-	properties([
-		[$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', numToKeepStr: '15']]
-	])
-	
-	stage('Checkout') {
-		checkout scm
+pipeline {
+	agent any
+
+	options {
+		buildDiscarder(logRotator(numToKeepStr:'15'))
 	}
-	
-	stage('Gradle Build') {
-		sh "./gradlew clean cleanGenerateXtext build createLocalMavenRepo -PuseJenkinsSnapshots=true -PJENKINS_URL=$JENKINS_URL -PcompileXtend=true -PignoreTestFailures=true --refresh-dependencies --continue"
-		step([$class: 'JUnitResultArchiver', testResults: '**/build/test-results/test/*.xml'])
+
+	tools { 
+		maven 'M3'
 	}
 	
-	stage('Maven Build') {
-		def mvnHome = tool 'M3'
-		env.M2_HOME = "${mvnHome}"
-		dir('.m2/repository/org/eclipse/xtext') { deleteDir() }
-		sh "${mvnHome}/bin/mvn -f releng --batch-mode --update-snapshots -Dmaven.repo.local=.m2/repository -DJENKINS_URL=$JENKINS_URL -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn clean install"
-	}
 	
-	archive 'build/**'
+	stages {
+		stage('Checkout') {
+			steps {
+				checkout scm
+			}
+		}
+
+		stage('Gradle Build') {
+			steps {
+				sh 'sh ./1-gradle-build.sh'
+				step([$class: 'JUnitResultArchiver', testResults: '**/build/test-results/test/*.xml'])
+			}
+		}
+		
+		stage('Maven Build') {
+			steps {
+				sh 'sh ./2-maven-build.sh'
+			}
+		}
+	}
+
+	post {
+		success {
+			archiveArtifacts artifacts: 'build/**'
+		}
+		failure {
+			slackSend baseUrl: 'https://itemis.slack.com/services/hooks/jenkins-ci/', botUser: true, channel: 'xtext-builds', color: '#FF0000', message: "FAILED: <${env.BUILD_URL}|${env.JOB_NAME}#${env.BUILD_NUMBER}>", token: '1vbkhv8Hwlp3ausuFGj1BdJb'
+		}
+		regression {
+			slackSend baseUrl: 'https://itemis.slack.com/services/hooks/jenkins-ci/', botUser: true, channel: 'xtext-builds', color: '#FF0000', message: "FAILED: <${env.BUILD_URL}|${env.JOB_NAME}#${env.BUILD_NUMBER}>", token: '1vbkhv8Hwlp3ausuFGj1BdJb'
```

Comment:
do you already know if this can be done via env as well?

Reply:
current version only uses `changed` and scripts within that. This is portable for both Jenkins versions.